R/zchunk_L131.nonco2_proc_R_S_T_Y.R

```diff
@@ -36,55 +39,168 @@ module_emissions_L131.nonco2_proc_R_S_T_Y_DISABLED <- function(command, ...) {
-    # There's also a `same_precursors_as(x)` you can use
-    # If no precursors (very rare) don't call `add_precursor` at all
-    tibble() %>%
-      add_title("descriptive title of data") %>%
-      add_units("units") %>%
-      add_comments("comments describing how data generated") %>%
-      add_comments("can be multiple lines") %>%
+    EPA_Ind <- get_data(all_data, "emissions/EPA_FCCC_IndProc_2005")
+
+    # EDGAR emissions data
+    get_data(all_data, "emissions/EDGAR_NH3") %>%
+      gather(year, value, -`IPCC-Annex`, -`World Region`, -ISO_A3, -Name, -IPCC, -IPCC_description) %>%
+      mutate(Non.CO2 = 'NH3') ->
+      EDGAR_NH3
+    get_data(all_data, "emissions/EDGAR_CH4") %>%
+      gather(year, value, -`IPCC-Annex`, -`World Region`, -ISO_A3, -Name, -IPCC, -IPCC_description) %>%
+      mutate(Non.CO2 = 'CH4')->
+      EDGAR_CH4
+    get_data(all_data, "emissions/EDGAR_N2O") %>%
+      gather(year, value, -`IPCC-Annex`, -`World Region`, -ISO_A3, -Name, -IPCC, -IPCC_description) %>%
+      mutate(Non.CO2 = 'N2O')->
+      EDGAR_N2O
+    EDGAR_VOC <- get_data(all_data, "emissions/EDGAR_NMVOC") %>%
+      gather(year, value, -`IPCC-Annex`, -`World Region`, -ISO_A3, -Name, -IPCC, -IPCC_description) %>%
+      mutate(Non.CO2 = 'NMVOC')->
+      EDGAR_VOC
+    get_data(all_data, "emissions/EDGAR_NOx") %>%
+      gather(year, value, -`IPCC-Annex`, -`World Region`, -ISO_A3, -Name, -IPCC, -IPCC_description) %>%
+      mutate(Non.CO2 = 'NOx')->
+      EDGAR_NOx
+    get_data(all_data, "emissions/EDGAR_SO2") %>%
+      gather(year, value, -`IPCC-Annex`, -`World Region`, -ISO_A3, -Name, -IPCC, -IPCC_description) %>%
+      mutate(Non.CO2 = 'SO2')->
+      EDGAR_SO2
+    get_data(all_data, "emissions/EDGAR_CO") %>%
+      gather(year, value, -`IPCC-Annex`, -`World Region`, -ISO_A3, -Name, -IPCC, -IPCC_description) %>%
+      mutate(Non.CO2 = 'CO')->
+      EDGAR_CO
+
+     ### Perform computations  ###
+
+     # First: Compute subsector share of sectoral emissions using EPA data
+
+     EPA_Ind %>%
+       left_join(EPA_tech, by = "Source_Category") %>%
+       rename(subsector = fuel) %>%
+       select(-EPA_Source_Category_Raw, -Source_Category) %>%   # Remove unnecessary columns
+       unique(.) %>%
+       gather(Non.CO2, value, -sector, -subsector) %>%  # make a new column called Non.co2
+       na.omit() %>%  # delete rows with NAs
+       group_by(sector, subsector, Non.CO2) %>% # group by sector, subsector and Non.CO2
+       summarise(x = sum(value)) -> # sum the values
+       L131.EPA_nonco2_indproc.mlt
+
+     # Second: Compute share of sectoral emissions in each subsector using EPA data
+     GCAM_sector_tech %>%
+       select(supplysector, subsector, stub.technology, EPA_agg_sector, EPA_agg_fuel_ghg, EDGAR_agg_sector) %>%
+       filter(EDGAR_agg_sector %in% c("industry_processes", "chemicals", "landfills", "wastewater","aerosols",
+                                      "metals","foams","solvents", "semiconductors")) %>%
```

Comment:
This string vector (`industry processes...semiconductors`) gets referenced several times, it looks like. Probably define in a constant (in this file) for convenience.